### PR TITLE
Remove unnecessary lock around gameData when getting technology front…

### PR DESCRIPTION
**Pull request by DanVanAtta**
…ier. The locking throws a lock exception when getting a screenshot, but has no impact.

Plus, remove a couple of empty javadocs and commented out code. #952

Fixes

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/977)

### CODE REVIEW

**Comment by RoiEXLab** [comment]
@DanVanAtta I think the reason to lock the gameData was to ensure, that all units do not change while the screenshot is being taken... Whatever...
I think we can merge here without problems

**Comment by ron-murhammer** [comment]
@DanVanAtta A little confused here. I would think we want to lock the data while grabbing the data to ensure things aren't changing?

**Comment by DanVanAtta** [comment]
It's just not necessary. The circumstances in which the technology data could change are esoteric and remote.

**Comment by ron-murhammer** [comment]
@DanVanAtta I'd have to think more about when and how often it can change (some maps use technologies in more creative ways) but I don't think this change would work at all given:
`  public TechnologyFrontier getTechnologyFrontier() {
    ensureLockHeld();
    return technologyFrontier;
  }`

**Comment by DanVanAtta** [comment]
Testing things out, screenshot works with this update.
The lock is broken, it is not held for long enough. We could easily have the game technology data change right after or before the lock is acquired. Furthermore, without sync blocks, you can't really guarantee that the lines will not be re-ordered. For this lock to be effective, we would need to lock the game data as soon as the snapshot operation started, and release it when the operation was finished.

**Comment by DanVanAtta** [comment]
Double checked that maps with techs work okay. That means a lock is held already when we enter getTechFrontier.

**Comment by ron-murhammer** [comment]
@DanVanAtta So I tested this in 1.9 and didn't get any errors so its something that's been introduced since then. My concern is while it appears to resolve the 'export screenshot' error popup in master, it could break other paths that call through 'getTechAdvances'.

This needs more investigation before making a change like this.

**Comment by DanVanAtta** [comment]
To protect the other code paths, we can remove the 'ensureLock'

The locking of stuff in GameData is cargo culting. Note it is inconsistent:

```

  /**
   * @return a collection of all units in the game
   */
  public UnitsList getUnits() {
    // ensureLockHeld();
    return unitsList;
  }
```
https://github.com/triplea-game/triplea/blob/master/src/games/strategy/engine/data/GameData.java#L143


The list of the other code paths looks good, does not look like anything in here even has a chance of a lot of contention:
techfrontierusages


Going back to the original commit, we can see that it was not added for any specific reason. It is just there.. Looking at those usages (` git log -p d889980574676a8223e64eae84236d0695b01a7b d889980574676a8223e64eae84236d0695b01a7b^
`)

### Usage number 1 - Gameparser
GameParser is single threaded, no locking is needed there.

### Usage number 2 - TechAdvance.java

```
        if( !data.getTechnologyFrontier().isEmpty()) {
       
          if(player!=null)
          {
            
            return player.getTechnologyFrontierList().getAdvances();
          }
          else 
          {
          
            return data.getTechnologyFrontier().getTechs(); 
          }
        }
```
We still have a race condition between player.getTechnologyFrontierList() and data.getTechnologyFrontier(). The locking does not help.


Second block
```
      if( player != null && !data.getTechnologyFrontier().isEmpty())
          return player.getTechnologyFrontierList().getFrontiers();
```
Again, race condition between player and `data`, the `data` could change as soon as we exit the if block. The lock was released to early to defend against that. Pretty much a similar story for the majority of the game data locking.

**Comment by DanVanAtta** [comment]
Second, you mentioned you could not repro with 1.9?  The exception is only maps that have no techs. Testing the other maps they are okay both with/without this patch.

Yes, this locking problem was something introduced, but the fix is not to go backwards.

**Comment by DanVanAtta** [comment]
@ron-murhammer let's get this wrapped up, it's the last thing known to be broken in 1.9. We'll have a new release candidate once this merged, and then we need time to do regression testing of that candidate. If this is wrapped up this soon, we can get the website links updated tonight and notify the community. That then gives a good week, and next weekend we can start the migration to 1.9 process.

**Comment by ron-murhammer** [comment]
@DanVanAtta Alright, I don't particularly like the change but agree the locking is inconsistent and probably in many cases unnecessary. Just concerned about changing things like this as there is lots of indirection and hard to follow code and I'm guessing some of the locking is needed. I also don't like that it works in 1.9 and somehow was broken since then. Since it is a pretty simple function of export a screenshot, you'd think we should have some sort of unit test so we would have realized when it broke.
